filebeat/input/filestream/fswatch_test.go

```diff
@@ -118,6 +119,58 @@ func setupFilesForScannerTest(t *testing.T, tmpDir string) {
 	}
 }
 
+func TestFileWatcherRenamedTruncated(t *testing.T) {
+	tmpDir := t.TempDir()
+
+	fs, err := newFileScanner([]string{filepath.Join(tmpDir, "app.log*")}, fileScannerConfig{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	w := fileWatcher{
+		interval:     1 * time.Second,
+		log:          logp.L(),
+		scanner:      fs,
+		events:       make(chan loginp.FSEvent),
+		sameFileFunc: os.SameFile,
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	go w.Run(ctx)
+
+	appLogPath := filepath.Join(tmpDir, "app.log")
+	rotatedAppLogPath := filepath.Join(tmpDir, "app.log.1")
+	err = ioutil.WriteFile(appLogPath, []byte("my longer log line"), 0o600)
```

Comment:
`ioutil` has been deprecated, you can use `os.WriteFile` instead:
```suggestion
	err = os.WriteFile(appLogPath, []byte("my longer log line"), 0o600)
```